clientv3/integration/kv_test.go

```diff
@@ -473,8 +473,8 @@ func TestKVNewAfterClose(t *testing.T) {
 
 	donec := make(chan struct{})
 	go func() {
-		if _, err := cli.Get(context.TODO(), "foo"); err != context.Canceled {
-			t.Fatalf("expected %v, got %v", context.Canceled, err)
```

Comment:
This is added because previously, no matter what, `retry.go` switches endpoint (resets connection) so didn't hit `grpc.ErrClientConnClosing` path. Now, only does switch when it's transient error, so `grpc.ErrClientConnClosing` is expected when concurrent goroutine closes client.